src/main/java/com/intuit/graphql/orchestrator/federation/EntityTypeMerger.java

```diff
@@ -39,6 +40,6 @@ private void merge(EntityMergingContext entityMergingContext) {
     private final String typename;
     private final String serviceNamespace;
     private final TypeDefinition baseType;
-    private final TypeDefinition typeExtension;
+    private final EObject typeExtension;
```

Comment:
Can this be TypeSystemDefinition? EObject is like Object class in the xtext world

Reply:
Lets discuss why is this needed